bin/prettier.js

```diff
@@ -42,9 +42,10 @@ if (!filenames.length && !stdin) {
 }
 
 function formatWithShebang(input) {
-  const programInput = input.split("\n");
-  const shebang = programInput.shift() + "\n";
-  return shebang + jscodefmt.format(programInput.join("\n"), options);
+  const index = input.indexOf("\n");
+  const shebang = input.slice(0, index);
+  const programInput = input.slice(index);
+  return shebang + "\n" + format(programInput, options);
```

Comment:
Nit, can you do slice(index + 1) and not reprint the \n. Otherwise looks great to me! Thanks for the update